libgrust/libproc_macro/proc_macro.cc

```diff
@@ -21,3 +21,32 @@
 // <http://www.gnu.org/licenses/>.
 
 #include "proc_macro.h"
+
+namespace Procmacro {
+
+Procmacro
+Procmacro::make_derive (char *trait_name, char **attributes, std::uint64_t size,
+			CustomDeriveMacro macro)
+{
+  ProcmacroPayload payload;
+  payload.custom_derive = {trait_name, attributes, size, macro};
```

Comment:
I'm wondering should you map this over to a std::string in the struct so we dont have to worry about doing things like strdup

Reply:
To be fair I'm not even sure we need the constructors, these structures should be embedded in the final proc_macro library and thus will be directly mapped to the file's content. The strings won't ever be created from us but rather by the backend from the trait name etc... in the ast.